django/thunderstore/repository/models/chunky.py

```diff
@@ -0,0 +1,119 @@
+import gzip
+import json
+from typing import List
+
+from django.db.models import CASCADE, PROTECT, ForeignKey, OneToOneField, QuerySet
+
+from thunderstore.community.models import Community, PackageListing
+from thunderstore.core.mixins import TimestampMixin
+from thunderstore.repository.cache import (
+    get_package_listing_queryset,
+    order_package_listing_queryset,
+)
+from thunderstore.storage.models import DataBlob, DataBlobGroup
+
+
+# TODO: where should this stuff reside in (not in the chunky.py for sure)
+# TODO: unit tests
+# TODO: add cronjob celery task that builds new caches and prunes old ones
+class APIV1ChunkedPackageCache(TimestampMixin):
+    community: Community = ForeignKey(
+        "community.Community",
+        related_name="chunked_package_list_cache",
+        on_delete=CASCADE,
+    )
+    index: DataBlob = OneToOneField(
+        "storage.DataBlob",
+        related_name="chunked_package_indexes",
+        on_delete=PROTECT,
+    )
+    chunks: DataBlobGroup = ForeignKey(
+        "storage.DataBlobGroup",
+        related_name="chunked_package_list_cache",
+        on_delete=PROTECT,
+    )
+
+    class Meta:
+        get_latest_by = "datetime_created"
```

Comment:
Perhaps it would make sense to create a classmethod for fetching the latest effective cache instead of querying it on the API view and assuming this implicitly gets applied? The interface for using this class would be more clear that way at least I think